### PR TITLE
backend/handlers: use handlers that can't return an error

### PR DESCRIPTION
**Pull request by benma**
The error returned by handlers is not propagated to the frontend in production. For errors that should be displayed to the user, we currently use the `{ success: false }` pattern, sometimes with an error message and/or error code.

It has been a frequent confusion that one can simply return an error from the handler. This commit adds a non-error version of the handler function, with the goal that all handlers eventually use this one, so errors are properly handled in all cases.

### CODE REVIEW

**Review by Beerosagos — CHANGES_REQUESTED** [review]
I think there are a couple  small changes needed

File: backend/devices/bitbox02/handlers/handlers.go [hunk 1]
```diff
@@ -119,72 +119,72 @@ func maybeBB02Err(err error, log *logrus.Entry) map[string]interface{} {
...
-	}, nil
+	}
 }
 
-func (handlers *Handlers) postSetDeviceName(r *http.Request) (interface{}, error) {
+func (handlers *Handlers) postSetDeviceName(r *http.Request) interface{} {
 	jsonBody := map[string]string{}
 	if err := json.NewDecoder(r.Body).Decode(&jsonBody); err != nil {
-		return nil, errp.WithStack(err)
+		return map[string]interface{}{"success": true}
```
[the remark this answers is not in the document]
Comment (benma) [reply backend/devices/bitbox02/handlers/handlers.go]: true! :stuck_out_tongue: fixed

File: frontends/web/src/components/qrcode/qrcode.tsx [hunk 2]
```diff
@@ -28,8 +28,8 @@ export const QRCode = ({
   data,
   size = 256,
 }: TProps) => {
-  const src = useLoad(data !== undefined ? getQRCode(data) : null, [data]);
-  if (!src) {
+  const qrCode = useLoad(data !== undefined ? getQRCode(data) : null, [data]);
+  if (!qrCode) {
```
[the remark this answers is not in the document]
Comment (benma) [reply frontends/web/src/components/qrcode/qrcode.tsx:32]: i don't think so - `qrCode` itself can be undefined. if it is defined, then either success=false or the data is available, both are handled below. the TS typechecker did well here!

[the remark this answers is not in the document]
Comment (Beerosagos) [reply frontends/web/src/components/qrcode/qrcode.tsx:32]: You're right! Don't know why I said that :D

**Pull request merged**